### PR TITLE
Step counter

### PR DESCRIPTION
**Pull request by ChrisDick**
# Description

update to imu API for the step counter released in 1.70

### CODE REVIEW

**Comment by ChrisDick** [comment]
is there a way to run the linters locally?

**Comment by npentrel** [comment]
Yes but not yet documented, feel free to just merge and I'll clean up later

**Review by npentrel — APPROVED** [review]

File: docs/tildagon-apps/reference/badge-hardware.md [hunk 1]
```diff
@@ -372,7 +372,7 @@ To use the `Pin`s:
 
 ## `IMU`
 
-The IMU device is a highly integrated, low power inertial measurement unit (IMU) that combines precise acceleration and angular rate (gyroscopic) measurement. The triple axis device has been configured to measure 2g and 2 degree per second ranges.
+The IMU device is a highly integrated, low power inertial measurement unit (IMU) that combines precise acceleration and angular rate (gyroscopic) measurement. The triple axis device has been configured to measure 2g and 2 degree per second ranges. It also has a step count function intended for wrist mounted applications
```
Comment (npentrel) [inline docs/tildagon-apps/reference/badge-hardware.md:375]:
```suggestion
The IMU device is a highly integrated, low power inertial measurement unit (IMU) that combines precise acceleration and angular rate (gyroscopic) measurement. The triple axis device has been configured to measure 2g and 2 degree per second ranges. It also has a step count function intended for wrist mounted applications.
```

File: docs/tildagon-apps/reference/badge-hardware.md [hunk 2]
```diff
@@ -412,6 +412,38 @@ class ExampleApp(app.App):
 __app_export__ = ExampleApp
 ```
```
Comment (npentrel) [inline docs/tildagon-apps/reference/badge-hardware.md:414]:
```suggestion

The following example app measures steps as you walk aroudn with your badge.

```

File: docs/tildagon-apps/reference/badge-hardware.md [hunk 3]
```diff
@@ -421,7 +455,7 @@ The api currently only allows access to the raw data.
 | ------ | ----------- | --------- | ------- |
 | `acc_read()` | Get the accelerometer data. | None | `(x,y,z)`: The accelerometer data as a tuple of floats (m/s^2). |
 | `gyro_read()` | Get the gyro data. | None | `(x,y,z)`: The gyro data as a tuple of floats (d/s). |
-
+| `step_counter_read()` | Get the step count | None | `count`: The step count |
```
Comment (npentrel) [inline docs/tildagon-apps/reference/badge-hardware.md:458]:
```suggestion
| `step_counter_read()` | Get the step count | None | `count`: The step count |

```

File: docs/tildagon-apps/reference/badge-hardware.md [hunk 4]
```diff
@@ -412,6 +412,40 @@ class ExampleApp(app.App):
...
+        else:
+            self.steps_read = imu.step_counter_read()
+
+    def draw(self, ctx):
+        ctx.save()
+        ctx.rgb(0.2,0,0).rectangle(-120,-120,240,240).fill()
+        if self.steps_read:
+            ctx.rgb(1,0,0).move_to(-80,-40).text("steps:\n{}\n".format(self.steps_read))
+        else:
+            ctx.rgb(1,0,0).move_to(-80,0).text("no readings yet")
```
Comment (npentrel) [inline docs/tildagon-apps/reference/badge-hardware.md:443]:
```suggestion
        ctx.rgb(0.2, 0, 0).rectangle(-120, -120, 240, 240).fill()
        if self.steps_read:
            ctx.rgb(1, 0, 0).move_to(-80, -40).text("steps:\n{}\n".format(self.steps_read))
        else:
            ctx.rgb(1, 0,  0).move_to(-80, 0).text("no readings yet")
```

File: docs/tildagon-apps/reference/badge-hardware.md [hunk 5]
```diff
@@ -412,6 +412,40 @@ class ExampleApp(app.App):
...
+            self.button_states.clear()
+            self.minimise()
+        else:
+            self.steps_read = imu.step_counter_read()
+
+    def draw(self, ctx):
+        ctx.save()
+        ctx.rgb(0.2, 0, 0).rectangle(-120, -120, 240, 240).fill()
+        if self.steps_read:
+            ctx.rgb(1, 0, 0).move_to(-80, -40).text("steps:\n{}\n".format(self.steps_read))
```
Comment (npentrel) [inline docs/tildagon-apps/reference/badge-hardware.md:441]:
```suggestion
            ctx.rgb(1, 0, 0).move_to(-80, -40).text(
                "steps:\n{}\n".format(self.steps_read))
```

File: docs/tildagon-apps/reference/badge-hardware.md [hunk 6]
```diff
@@ -412,6 +412,41 @@ class ExampleApp(app.App):
...
+        ctx.save()
+        ctx.rgb(0.2, 0, 0).rectangle(-120, -120, 240, 240).fill()
+        if self.steps_read:
+            ctx.rgb(1, 0, 0).move_to(-80, -40).text(
+                "steps:\n{}\n".format(self.steps_read))
+        else:
+            ctx.rgb(1, 0,  0).move_to(-80, 0).text("no readings yet")
+        ctx.restore()
+
+__app_export__ = ExampleApp
```
Comment (npentrel) [inline docs/tildagon-apps/reference/badge-hardware.md:447]:
```suggestion

__app_export__ = ExampleApp
```

**Pull request merged**